pkg/auth/providers/azure/config.go

```diff
@@ -24,17 +28,37 @@ func graphEndpointDeprecated(endpoint string) bool {
 }
 
 func updateAzureADConfig(c *v32.AzureADConfig) {
-	// Update the Graph Endpoint.
-	graphEndpointMigration := map[string]string{
-		globalAzureADGraphEndpoint: globalMSGraphEndpoint,
-		chinaAzureADGraphEndpoint:  chinaMSGraphEndpoint,
+	if isConfigForChina(c) {
+		updateConfigForChina(c)
+	} else {
+		updateConfigForGlobal(c)
 	}
-	// This will be a no-op if the config already uses the new endpoint.
-	if newEndpoint, oldUsed := graphEndpointMigration[c.GraphEndpoint]; oldUsed {
-		c.GraphEndpoint = newEndpoint
+}
+
+func isConfigForChina(c *v32.AzureADConfig) bool {
+	return strings.HasSuffix(c.GraphEndpoint, ".cn") || strings.HasSuffix(c.GraphEndpoint, ".cn/")
+}
+
+func updateConfigForGlobal(c *v32.AzureADConfig) {
+	// Update the Graph Endpoint.
+	if c.GraphEndpoint == globalAzureADGraphEndpoint {
+		c.GraphEndpoint = globalMSGraphEndpoint
+		// Update the Auth Endpoint and Token Endpoint.
+		c.AuthEndpoint = fmt.Sprintf("%s%s/oauth2/v2.0/authorize", c.Endpoint, c.TenantID)
+		c.TokenEndpoint = fmt.Sprintf("%s%s/oauth2/v2.0/token", c.Endpoint, c.TenantID)
 	}
+}
```

Comment:
Use an early return.
```suggestion
func updateConfigForGlobal(c *v32.AzureADConfig) {
	// Update the Graph Endpoint.
	if c.GraphEndpoint != globalAzureADGraphEndpoint {
	    return
	}
	c.GraphEndpoint = globalMSGraphEndpoint
	// Update the Auth Endpoint and Token Endpoint.
	c.AuthEndpoint = fmt.Sprintf("%s%s/oauth2/v2.0/authorize", c.Endpoint, c.TenantID)
	c.TokenEndpoint = fmt.Sprintf("%s%s/oauth2/v2.0/token", c.Endpoint, c.TenantID)
}
```